Sources/Kafka/KafkaConsumer.swift

```diff
@@ -334,22 +355,32 @@ public final class KafkaConsumer: Sendable, Service {
         while !Task.isCancelled {
             let nextAction = self.stateMachine.withLockedValue { $0.nextPollLoopAction() }
             switch nextAction {
-            case .pollForAndYieldMessage(let client, let source):
-                let events = client.eventPoll()
-                for event in events {
-                    switch event {
-                    case .consumerMessages(let result):
-                        // We do not support back pressure, we can ignore the yield result
-                        _ = source.yield(result)
-                    default:
-                        break // Ignore
+            case .pollForEventsAndMessage(let client, let source):
+                // Event poll to serve any events queued inside of `librdkafka`.
+                _ = client.eventPoll()
+                // Poll for new consumer message.
+                var result: Result<KafkaConsumerMessage, Error>?
+                do {
+                    if let message = try client.consumerPoll() {
```

Comment:
It seems that previously we would poll for up to 100 messages before sleep and it looks like now it would be just one message.

Reply:
If I read this correctly we are polling a single message and then yielding a single message right? It would probably be better if we read them in batches of 100 and then yield all of them at once. Would mean we have to acquire the locks a lot less.

Reply:
Not sure it is entirely what I mean. This comment was related to changes before introducing task group.
There was a sleep after every consumed message.
So, it is not relevant now except probably this part: https://github.com/swift-server/swift-kafka-client/pull/139/files#r1360663871